common/src/main/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManager.java

```diff
@@ -200,20 +200,22 @@ public void run() {
       } catch (InterruptedException e) {
         running = false;
       }
-      try {
-        if (running && needsReload()) {
-          try {
-            trustManagerRef.set(loadTrustManager());
-            this.reloadCount += 1;
-          } catch (Exception ex) {
-            logger.warn(
-                "Could not load truststore (keep using existing one) : " + ex.toString(), ex);
+      synchronized (this) {
```

Comment:
Why do we need this `synchronized` ?

Reply:
@mridulm, this change follows [NonAtomicVolatileUpdate](https://errorprone.info/bugpattern/NonAtomicVolatileUpdate).

Reply:
This change is not required IMO. These `volatile` variables being updated only in this method.
Technically, they dont need to be `volatile` - but were made volatile only to ensure we can mutate them from tests, and they become visible.

Note that if the intent is to actually address the corner case race condition of mutation from the test (that is reasonable) - then we should wrap the mutations from `ReloadingX509TrustManagerSuiteJ` as well in `synchronized`